bokeh/core/property/color.py

```diff
@@ -127,6 +127,11 @@ def __init__(self, default=Undefined, help=None):
                        r"\s*([01]\.?\d*?)\)"),
                  Regex(r"^rgb\(((25[0-5]|2[0-4]\d|1\d{1,2}|\d\d?)\s*,"
                        r"\s*?){2}(25[0-5]|2[0-4]\d|1\d{1,2}|\d\d?)\s*?\)"),
+                 Regex(
+                     r"^hsl(a)?\((((360|3[0-5]\d|2\d{1,2}|1\d{1,2}|\d\d?)(deg)?"
+                     r"|(400|3\d{1,2}|2\d{1,2}|1\d{1,2}|\d\d?)grad|"
+                     r"\d\.?\d*rad|[0-1]\.?\d*turn)(,?\s*(100|\d{1,2}%)){2}"
+                     r"(\s*\/\s*(1|0?\.?\d*))?\))"),
```

Comment:
Just for my own info, is there any official provenance for this regex?

Reply:
This needs to be derived from https://www.w3.org/TR/css-color-4/#the-hsl-notation. Note that there are two syntaxes in CSS4 spec, modern and legacy, that need to be implemented. I would simplify this and turn it into either a rudimentary parser combinator, or use a parser combinator library.

Reply:
@bryevdv no, I created it based on my knowledge of the hsl notation. I will review @mattpap comment and simplify this part

Reply:
@mattpap can we talk about it in slack?

Reply:
@bryevdv @mattpap since HSL notation has been deprecated, I think we can simplify this by just checking if the string starts with `hsl(` or `hsla(` and defer it. What do you guys think?

Reply:
More specifically, the `HSL` object has been deprecated, and that is because in general the actual color objects under `bokeh.colors` have really never seen much/any usage. The reason we would want to keep a more thorough validation here is so that we can alert users to a problem immediately (in Python) instead of just some browser JS console error (that many users may never see or know how to find). 

*That said*, it's pretty unlikely that many or any users will use HSLa strings either. We should not over-expend effort on this. I think there are some simpler HSLa regexes floating around the internet. I would suggest trying to use one of those and then making sure that unit tests properly cover a basic selection of common HSLa strings.